.github/workflows/ci.yml

```diff
@@ -221,6 +221,8 @@ jobs:
     - name: Log in to the Container registry
       if: ${{ matrix.variant.library == 'ruby' }}
       run: echo ${{ secrets.GITHUB_TOKEN }} | docker login ${{ env.REGISTRY }} -u ${{ github.actor }} --password-stdin
+    - name: Build proxy
```

Comment:
We do not build it, on purpose, to reduce CI time.

Reply:
I understand the desire to reduce CI time, that is why I made sure the `buildx build` commands are aligned:

- thanks to `--build-arg BUILDKIT_INLINE_CACHE=1` as soon as a new image that was built with the flag is published it will take 0s when there is no change from the previously published image (may need to use `--cache-from`)
- even as is, [It takes 5s](https://github.com/DataDog/system-tests/actions/runs/5475850663/jobs/9972604291?pr=1109#step:10:15) which is dwarfed by weblog image building by an order of magnitude.

Reply:
> But the question is : why should we want to rebuild it

Precisely for the reason highlighted by the failure this fixes: when sources of the docker image change it uses whatever has been published before instead of the current branch state, resulting in temporal inconsistency.

> Most of PR only run default scenario, and take 2 to 3 minutes

I am not sure I follow your point:

- once a new image built with inline cache is pushed it will be a noop (that also ensures temporal consistency as outlined above)
- agent and weblog images are rebuilt each time (for similar temporal consistency reasons) even for default scenarios, and _these_ are dwarfing proxy rebuild

Reply:
The point is that the source of the proxy are very stable: very few deps. It's all the point about publishing docker image to docker hub. If we need to update the source, we'll update the version number inside the tag. `datadog/system-tests:proxy-v0` we stay compatible with the current main branch and all PRs. If we need to update it, we'll bump to `datadog/system-tests:proxy-v1`. I've just pushed the process to update it on #1379. 

> agent and weblog images are rebuilt each time (for similar temporal consistency reasons) even for default scenarios, and these are dwarfing proxy rebuild

Because their sources are far less stable by several order of magnitude. For the `dev` part of our CI, we even need to get the very last artifact from all components. Without talking about the fact that we have lot of PR that modifies weblogs.

It's not the case for the proxy. So even if it's a small win, it's still a win. And having the image published on docker hub working with the main branch will be part of the contract, so no worry about that. If you're afraid of issue on that part, you can rebuild it on your CI. But as I want to offer this contract, this CI must use this feature (image on docker hub is up. to date).

Reply:
How do you suggest we proceed?

AIUI the process would be:

1. revert https://github.com/DataDog/system-tests/commit/fbb03ac20da6fce8a11090fd4443574ce6b6965e
2. since the PR uses `datadog/system-tests:proxy-v0` and [`utils/build/docker/proxy.Dockerfile`](https://github.com/DataDog/system-tests/blob/ace288bf8281ccc923a9dab05dd20def5a9c2c80/utils/build/docker/proxy.Dockerfile) changed, this PR's CI will fail
3. checkout PR branch
4. change `datadog/system-tests:proxy-v0` to `datadog/system-tests:proxy-v1`
5. commit and push to PR
7. since the PR references a nonexistent `datadog/system-tests:proxy-v1` this PR's CI will fail
8. build proxy image locally
9. push image as `datadog/system-tests:proxy-v1`
10. manually trigger a CI re-run for this PR
11. wait for CI to be green
12. merge

Reply:
1 to 5 -> yep

Then, I've added an optional step in the CI to build it if needed : 

6. add the label `build-proxy-image` -> it will make the CI rebuild the CI image locally (see this [doc](https://github.com/DataDog/system-tests/blob/main/docs/edit/update-docker-images.md), I made it few days ago)
7. ensure CI becomes green, got threw the review process
8. I will push image as `datadog/system-tests:proxy-v1` (my team is the owner of `datadog/system-tests` repo on hub.docker.com)
9. merge

Reply:
Done